tfstate/remote_tfe.go

```diff
@@ -0,0 +1,66 @@
+package tfstate
+
+import (
+	"context"
+	"io"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/pkg/errors"
+)
+
+func readTFEState(config map[string]interface{}, ws string) (io.ReadCloser, error) {
+
+	hostname, organization, token := *strpe(config["hostname"]), *strp(config["organization"]), *strpe(config["token"])
+
+	workspaces, ok := config["workspaces"].(map[string]interface{})
+	if !ok {
+		return nil, errors.Errorf("failed to parse workspaces")
+	}
+
+	name, prefix := *strpe(workspaces["name"]), *strpe(workspaces["prefix"])
+	if name != "" {
+		return readTFE(hostname, organization, name, token)
+	}
+
+	if prefix != "" {
+		return readTFE(hostname, organization, prefix+ws, token)
+	}
+
+	return nil, errors.Errorf("workspaces requires either name or prefix")
+}
+
+func readTFE(hostname string, organization string, ws string, token string) (io.ReadCloser, error) {
+	var address string
+	address = tfe.DefaultAddress
+	if hostname != "" {
+		address = "https://" + hostname
+	}
+
+	var err error
+	var client *tfe.Client
+	if token != "" {
+		client, err = tfe.NewClient(&tfe.Config{
+			Address: address,
+			Token:   token,
+		})
+	} else {
+		client, err = tfe.NewClient(&tfe.Config{
+			Address: address,
```

Comment:
In this case, use the environment variable `TFE_TOKEN`.
https://registry.terraform.io/providers/hashicorp/tfe/latest/docs

Reply:
The TFE provider is different than using Terraform Cloud / Enterprise as a backend. The `TFE_TOKEN` environment variable still works on the provider, but not the backend configuration. 

See for more info: https://github.com/hashicorp/terraform/issues/21965